test/Microsoft.Extensions.Configuration.Test/ConfigurationTest.cs

```diff
@@ -92,14 +92,65 @@ public void AsEnumerateFlattensIntoDictionaryTest()
             var config = configurationBuilder.Build();
 
             var dict = config.AsEnumerable().ToDictionary(k => k.Key, v => v.Value);
-            Assert.Equal("ValueInMem1", config["Mem1:KeyInMem1"]);
-            Assert.Equal("ValueDeep1", config["Mem1:KeyInMem1:Deep1"]);
-            Assert.Equal("ValueInMem2", config["Mem2:KeyInMem2"]);
-            Assert.Equal("ValueDeep2", config["Mem2:KeyInMem2:Deep2"]);
-            Assert.Equal("ValueInMem3", config["Mem3:KeyInMem3"]);
-            Assert.Equal("ValueDeep3", config["Mem3:KeyInMem3:Deep3"]);
+            Assert.Equal("ValueInMem1", dict["Mem1:KeyInMem1"]);
+            Assert.Equal("ValueDeep1", dict["Mem1:KeyInMem1:Deep1"]);
+            Assert.Equal("ValueInMem2", dict["Mem2:KeyInMem2"]);
+            Assert.Equal("ValueDeep2", dict["Mem2:KeyInMem2:Deep2"]);
+            Assert.Equal("ValueInMem3", dict["Mem3:KeyInMem3"]);
+            Assert.Equal("ValueDeep3", dict["Mem3:KeyInMem3:Deep3"]);
         }
 
+        [Fact]
+        public void AsEnumerateStripsKeyFromChildren()
```

Comment:
config != section guards against that, but, I think that does reveal a subtle issue where we no longer display the value of the current node as a result